[28기 진민욱][Add] Login, Main 페이지 완성 & 리팩토링 중 & 추가 기능 구현 중

minukk opened the pull request:
-Login 페이지 레이아웃 완성
-Main 페이지 레이아웃 완성
-메인 페이지 댓글 기능 완성
-id,pw validation 추가

## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [x] 리팩토링
- [ ] 버그 수정
- [x] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 메인 페이지 댓글 기능 로직 구현 
-로그인 페이지 id, pw 유효성 검사 구현
<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. 댓글 입력시 댓글폼에 입력한 댓글 추가
- 입력창에 입력이 끝나고 버튼을 클릭할 시 입력창에 있는 값이 댓글 폼에 추가된다.
-테스트 방법: 입력창에 문자열을 입력하고 게시 버튼을 클릭, 버튼을 클릭할 시 입력창의 문자열이 댓글 폼에 정상적으로 추가되는가
2. 로그인 페이지에서 id, pw 입력시 정규 표현식을 사용하여 검사한 후 검사를 통과하면 로그인 버튼 활성화.

<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
-HTML 시멘틱의 이해
-레이아웃의 구조 이해
-'어떻게 작성하는 것이 가독성이 좋을 것인가'에 대한 고민
-'리팩토링을 어떻게 하는 것이 코드를 더 깔끔하게 만들것인가'에 대한 고민
-이벤트 전파( 버블링, 캡처)의 개념을 어떻게 사용하는지 알 수 있었음

<br />

## :: 기타 질문 및 특이 사항

joonsikyang commented on students/14th/jihyeryu/js/main.js:
```diff
@@ -0,0 +1,40 @@
+
+    if ((commentInputSide.value !== '' && e.target == commentInputBtn) || (commentInputSide.value !== '' && e.keyCode === 13)) {
+
+        const commentGroup = document.getElementById('comment-group')
+        const div = document.createElement('div');
+        commentGroup.appendChild(div);
+        div.classList.add('user-account-name')
+        div.innerHTML += `  <div class="mensions-list">
+        <span><b>ryuwisdom</b>
+            &nbsp;${commentInputSide.value}</span>
+    </div>
+    <div class="mention-like">
+        <img src="./image/heart.png" alt="">
+    </div>
+        `
```
- 전체 코드를 살펴보면 innerHTML을 사용해서 훨씬 효율적으로 짤 수 있겠네요! 
- check 함수와 inputComment 함수 분리 Good!

ryuwisdom replied:
전체 코드를 innerHTML로 묶는게 더 수월하다는 뜻이 맞나요..? 
레플릿 풀면서 익혔던 appendChild()/ createElement()를 일부러 적용해보면서 조금 복잡하다는 생각은 들었어요 ,
피드백 주신 내용대로 전체를 innerHTML로 묶으니 코드가 확 가벼워졌어요!

joonsikyang commented on students/14th/jihyeryu/js/main.js:
```diff
@@ -0,0 +1,40 @@
+        <img src="./image/heart.png" alt="">
+    </div>
+        `
+        console.log("입력");
```
console.log 삭제!

ryuwisdom replied:
반영하였습니다😀

joonsikyang commented on students/14th/jihyeryu/js/main.js:
```diff
@@ -0,0 +1,40 @@
+
+}
+
+commentInputSide.addEventListener('keyup', checkCommnet);
+commentInputSide.addEventListener('keydown', inputComment);
```
keyup, keydown 따로 주기 보다는 keyup 이벤트 동작하면 무조건  엔터가 입력된건지 아닌지 확인하고, 엔터라면 inputComment 함수를 실행시키는 로직도 생각해보시기 바랍니다.

ryuwisdom replied:
반영해보았습니다. 
수정하면서 코드를 짤 때 좀 더 고심해보게되었어요😆!

joonsikyang commented on students/14th/jihyeryu/style/login.css:
```diff
@@ -0,0 +1,89 @@
+}
+
+body {
+    height: 100%;
+    width: 100%;
```
참고로 vh, vw 단위도 있습니다 :) 화면 전체를 차지하게 하고 싶을 때는 vh, vw 단위를 많이 사용하기도 합니다.

ryuwisdom replied:
반영해보았습니다.👐  블로그 정리해야겠습니다..!

joonsikyang commented on students/14th/jihyeryu/style/login.css:
```diff
@@ -0,0 +1,89 @@
+    background: white;
+    position: relative;
+}
+.logo {
+    width: 250px;
+    height: 70px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    position: absolute;
+    top: 50px;
+}
+.logo-img {
+    width: 175px;
+}
+.main-box .login {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    flex-direction: column;
+    position: absolute;
+    top: 150px;
+}
```
가독성을 위해 요소와 요소 사이는 한 줄 띄워주세요!

ryuwisdom replied:
반영하였습니다😀

joonsikyang commented on students/14th/jihyeryu/style/login.css:
```diff
@@ -0,0 +1,89 @@
+    position: absolute;
+    top: 340px;
+}
+.forgetpw p {
```
한 줄 띄워주세요~

ryuwisdom replied:
반영하였습니다😀

joonsikyang commented on students/14th/jihyeryu/style/main.css:
```diff
@@ -0,0 +1,476 @@
+    width: 100%;
+    height: 58px;
+}
+.other-profile {
+    width: 100%;
+    height: 58px;
+    display: flex;
+    justify-content: flex-start;
+    align-items: center;
+}
+.other-profile-photo {
+    width: 32px;
+    height: 32px;
+    margin-right: 10px;
+}
+
+.other-profile-photo img {
+    width: 32px;
+    height: 32px;
+    border-radius: 50%;
+    object-fit: cover;
+}
+.follow-btn {
+    width: 40px;
+}
+.follow-btn button {
+    color:#0095f6;
+    outline: none;
+    border: none;
+    background-color: transparent;
+    cursor: pointer;
+}
+.intro-menu {
+    width: 293px;
+    height: 136px;
+}
+.intro-menu p {
+    font-size: 11px;
+    color: #c7c7c7;
+}
```
전체적으로 naming도 그렇고 css 관련해서 고민해보신 흔적이 많이 보여서 좋네요! 요소와 요소 사이에 한 줄 띄워주는 것 수정해주시고, 나중에 sass 배우시면 어떻게 효율적으로 css 코드가 관리될 수 있는지 체험해보시면 좋겠습니다!

ryuwisdom replied:
네!! 🐥

joonsikyang commented on students/14th/jihyeryu/main.html:
```diff
@@ -0,0 +1,263 @@
+                    </div>
+                    <div class="icon">
+                        <ul>
+                            <li class="icon-menu">
+                                <img src="./image/home.png" alt="home">
+                            </li>
+                            <li class="icon-menu">
+                                <img src="./image/direct.png" alt="direct">
+                            </li>
+                            <li class="icon-menu">
+                                <img src="./image/explore.png" alt="explore">
+                            </li>
+                            <li class="icon-menu">
+                                <img src="./image/heart.png" alt="heart">
+                            </li>
+                            <li class="user icon-menu">
+                                <img src="./image/profile.jpeg" alt="profile">
+                            </li>
```
이렇게 반복되는 요소는 배열 데이터 타입을 활용하면 좋겠네요! 배열의 요소를 객체로 만들어서 데이터를 만들어 준 후 map 함수 적용해보시면 코드가 훨씬 짧고 효율적으로 바뀔 수 있습니다 ! 시도해보세요 :)

ryuwisdom replied:
반영해보았습니다!
map에 대해 알고있다 생각했는데 다시 해보니 너무 새로웠습니다..🤭  신기해요..

joonsikyang commented on students/14th/jihyeryu/main.html:
```diff
@@ -0,0 +1,263 @@
+    <script src="js/main.js"></script>
+</body>
+
+</html>
```
- p 태그가 계속 보이는데 활용법에 대해 좀 더 고민해보시면 좋을 것 같습니다.
- 부모 자식의 관계가 잘 드러날 수 있도록 들여쓰기 수정해주세요!
- 반복되는 img 태그의 alt 값을 전부 다르게 할 필요는 없습니다! 참고해주세요.

ryuwisdom replied:
p 태그 사용.. 고민해보겠습니다...😬
코드 간격 수정/반영해보았습니다. 혹시 추천하시는  formatter가 있나요? (저는 현재 beautify사용중입니다. ) 
alt값에 대한 부담이 적어졌어요! 감사합니다.

joonsikyang reviewed (commented):
지혜님! 수고하셨습니다👐 리뷰 내용 확인해주시고 리팩토링 해보시기 바랍니다 :)

ryuwisdom commented on pull_request_template.md:
```diff
@@ -2,19 +2,22 @@
 ## 기타 질문 및 특이 사항
-(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
-- 함수 로직이 너무 긴 것 같습니다. 좀 더 효율적인 방법이 없을지 궁금합니다.(예시) 
+질문 : [CSS] 어떤 경우에 width, height속성을 %로 조정하는지 궁금합니다.
+
+- 작업하면서 느꼈던 개선할 점
+  - 불명확한 id, class명 
+  - 시맨틱 태그 사용 부족, div태그 과다 사용
+  
```
준식님, 상세한 답변 감사합니다! 😀
프로젝트 진행하면서 보완해나가겠습니다.

ryuwisdom commented on students/14th/jihyeryu/js/login.js:
```diff
@@ -0,0 +1,17 @@
+const pw = document.getElementById("pw");
+const btn = document.getElementById("login-btn");
+
+function checkLogin (){
+    if(id.value !=="" && pw.value !==""){
+        console.log("+_+");
+        btn.style.backgroundColor='#0095f6';
+    }else {
+        btn.style.backgroundColor='rgb(201, 224, 249)';
+        console.log("*_*");
+    }
+}
```
반영하였습니다. 감사합니다.😀

ryuwisdom commented on students/14th/jihyeryu/js/login.js:
```diff
@@ -0,0 +1,17 @@
+    }
+}
+
+id.addEventListener('keyup', checkLogin);
+pw.addEventListener('keyup', checkLogin);
```
반영하였습니다. id, pw의 부모요소에 이벤트를 다는 것은 생각지도 못했습니다.! 감명받았어요🤭

ryuwisdom commented on students/14th/jihyeryu/login.html:
```diff
@@ -0,0 +1,31 @@
+  <body>
+    <section class="main">
+        <div class="main-container">
+          
```
반영하였습니다. 감사합니다.😀

ryuwisdom commented on students/14th/jihyeryu/login.html:
```diff
@@ -0,0 +1,31 @@
+     <link href="style/login.css" rel="stylesheet" type="text/css" />
+  </head>
+  <body>
+    <section class="main">
```
수정/반영하였습니다😀

ryuwisdom commented on students/14th/jihyeryu/login.html:
```diff
@@ -0,0 +1,31 @@
+                    <button id="login-btn">로그인</button>
+                </div>
+                <div class="forgetpw">
+                    <p>비밀번호를 잊으셨나요?</p>
```
<p>태그는 하나의 문단을 구성할때 사용합니다. + 블록요소로 한 줄 전체를 차지합니다.
<a>태그로 수정해보았습니다....😶
( a (anchor, 앵커) 요소는 웹문서 안에 있는 이미지 및 텍스트 등의 요소를 통해 다른 장소로 이동시키기 위한 링크를 걸 때 사용한다.)

ryuwisdom commented on students/14th/jihyeryu/login.html:
```diff
@@ -0,0 +1,31 @@
+                    <p>비밀번호를 잊으셨나요?</p>
+                </div>
+            </div>
+              </div>
```
반영하였습니다😀

ryuwisdom commented on students/14th/jihyeryu/js/main.js:
```diff
@@ -0,0 +1,40 @@
+const commentInputBtn = document.getElementById('comment-input-btn');
+const displayComment = document.getElementById('comment-side');
+const commentInputSide = document.getElementById('comment-input-side');
+
+
+
```
반영하였습니다😀

ryuwisdom commented on students/14th/jihyeryu/js/main.js:
```diff
@@ -0,0 +1,40 @@
+
+
+
+function checkCommnet() {
+    if (commentInputSide.value !== '') {
+        commentInputBtn.style.color = '#0095f6';
+    } else {
+        commentInputBtn.style.color = 'rgb(201, 224, 249)';
+    }
+}
```
수정/반영하였습니다😀  오타.. 주의하겠습니다😶 !!

minukk committed:
[Add] Login 페이지 레이아웃 & 유효성 검사 코드 작성 중
-Login 페이지 레이아웃 완성
-메인 페이지에서 사용될 로직 작성 중
-유효성 검사 작성 중

minukk committed:
[Add]메인 페이지 댓글 기능 추가 & 레이아웃 구조 변경
-메인 페이지의 댓글 기능 추가
-메인 페이지의 레이아웃 구조 변경

minukk committed:
[Modify] 메인 페이지 댓글 기능 수정
-메인 페이지의 댓글 기능 로직을 수정하였음.

minukk committed:
[Modify] 메인 페이지 댓글 기능 로직 수정 & CSS 변경
-메인 페이지의 댓글 기능 로직을 변경하였음
-input의focus를 리셋하였던 reset.css를 변경하였음

minukk committed:
[Modify] Login 버튼 활성화 로직 수정
-Login 버튼 활성화를 위해 button의 disabled attribute를 사용

minukk committed:
[Add] Login 페이지 버튼 hover 시 그림자 효과 추가
-Login 페이지 로그인 버튼 hover 시 box-shadow 효과 추가

minukk committed:
[Refactor] Main 페이지 리팩토링
-Main 페이지 CSS 리팩토링
-Main 페이지 JS 리팩토링

minukk committed:
[Add] 로그인 페이지id, password validation 추가
-id validation을 정규표현식을 사용하여 검사
-password validation을 정규표현식을 사용하여 검사

minukk committed:
[Add] Main 페이지 댓글 좋아요 기능 추가 중
-Main 페이지 input 으로 생성한 댓글 좋아요 기능 추가중

minukk committed:
[Modify] Main 페이지 댓글 좋아요 기능 수정
-이벤트 캡처의 개념으로 상위 요소의 이벤트에서 하위 요소의 이벤트를 감지
-하위 요소의 이벤트를 감지한 후 class 추가를 통해 색을 주입/삭제

minukk committed:
[Add] Main 페이지 댓글 삭제 기능 추가 & 댓글 좋아요 기능 수정
-댓글 좋아요 기능 수정
-댓글 삭제 기능 추가

minukk committed:
[Refact] Login 페이지 리팩토링
-로그인 페이지의 함수들을 되도록이면 한 가지 기능만 담당하도록 리팩토링
-로그인 페이지의 css 레이아웃을 flex로 변경(중앙에 균형이 맞게 맞추기 위해)
-(그 전에는 margin으로 정확히 중앙에 오지는 않았음.)

minukk committed:
[Refact] 메인 페이지 리팩토링

minukk committed:
[Modify] 로그인 페이지 수정 & 메인 페이지 검색 기능 추가 중
-메인 페이지 검색 기능 추가 중
